src/test/java/com/neo4j/docker/neo4jserver/TestMounting.java

```diff
@@ -240,6 +242,9 @@ void testCantWriteIfSecureEnabledAndNoPermissions_logs() throws IOException
 	void canMountAllTheThings_fileMounts(boolean asCurrentUser) throws Exception
 	{
 		Path testOutputFolder = HostFileSystemOperations.createTempFolder( "mount-everything-" );
+		Path confFile = Paths.get( "src", "test", "resources", "confs", "MountConf.conf" );
+		Files.copy( confFile, testOutputFolder.resolve( "neo4j.conf" ) );
+
 		try(GenericContainer container = setupBasicContainer( asCurrentUser, false ))
 		{
 			HostFileSystemOperations.createTempFolderAndMountAsVolume( container, "conf", "/conf", testOutputFolder );
```

Comment:
in this test it creates a folder `/tmp/mount-everything-####/` then underneath on this line (L250) it creates a subfolder  `/tmp/mount-everything-####/conf` which is where your conf file needs to go. At the moment you're not mounting it in a location where docker will see the conf.